tests/foreman/cli/test_discoveryrule.py

```diff
@@ -520,58 +386,18 @@ def test_negative_update_name(self, name, discoveryrule_factory):
         :parametrized: yes
         """
         rule = discoveryrule_factory()
-        with pytest.raises(CLIReturnCodeError):
-            DiscoveryRule.update({'id': rule.id, 'name': name})
-
-    @pytest.mark.tier3
-    def test_negative_update_hostname(self, discoveryrule_factory):
-        """Update discovery rule host name using number as a value
-
-        :id: c382dbc7-9509-4060-9038-1617f7fef038
-
-        :expectedresults: Rule host name is not updated
-
-        :CaseImportance: Medium
-
-        :CaseLevel: Component
-        """
-        rule = discoveryrule_factory()
-        with pytest.raises(CLIReturnCodeError):
-            DiscoveryRule.update({'id': rule.id, 'hostname': '$#@!*'})
-
-    @pytest.mark.tier3
-    def test_negative_update_limit(self, discoveryrule_factory):
-        """Update discovery rule host limit using invalid values
-
-        :id: e3257d8a-91b9-406f-bd74-0fd1fb05bb77
-
-        :expectedresults: Rule host limit is not updated
-
-        :CaseLevel: Component
-
-        :CaseImportance: Medium
-        """
-        rule = discoveryrule_factory()
-        host_limit = gen_string('alpha')
-        with pytest.raises(CLIReturnCodeError):
-            DiscoveryRule.update({'id': rule.id, 'hosts-limit': host_limit})
-
-    @pytest.mark.tier3
-    def test_negative_update_priority(self, discoveryrule_factory):
-        """Update discovery rule priority using invalid values
-
-        :id: 0778dd00-aa19-4062-bdf3-752e1b546ec2
-
-        :expectedresults: Rule priority is not updated
-
-        :CaseLevel: Component
-
-        :CaseImportance: Medium
-        """
-        rule = discoveryrule_factory()
         priority = gen_string('alpha')
+        host_limit = gen_string('alpha')
         with pytest.raises(CLIReturnCodeError):
-            DiscoveryRule.update({'id': rule.id, 'priority': priority})
+            target_sat.cli.DiscoveryRule.update(
+                {
+                    'id': rule.id,
+                    'name': name,
+                    'hostname': '$#@!*',
+                    'hosts-limit': host_limit,
+                    'priority': priority,
+                }
+            )
```

Comment:
Merging these four tests into one does not seem correct to me.

CLI tells us that only one parameter has an invalid value. Therefore this test will verify that CLI does not like/refuses only that specific argument.


```
[D 230801 07:00:19 hosts:99] testhost.com executing command: b'LANG=en_US.UTF-8  hammer -v -u admin -p guess --output=csv discovery-rule update --id="88" --name="" --hostname="$#@!*" --hosts-limit="bjSxIiYmwr" --priority="sqPvGlcyFf" '
[D 230801 07:00:20 hosts:101] testhost.com command result:
    stdout:
    
    stderr:
    (134, b"Could not update the rule:\n  Error: Option '--priority': Numeric value is required..\n  \n  See: 'hammer discovery-rule update --help'.\n")
    status: 64
```

I suggest keeping the tests as they were or removing them entirely.

Reply:
You are correct, instead of removing I am doing some randomization here.